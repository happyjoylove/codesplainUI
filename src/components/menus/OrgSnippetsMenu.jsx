import React, { PropTypes } from 'react';
import { MenuItem } from 'material-ui';
import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right';

import SnippetList from './SnippetList';
import CustomPropTypes from '../../util/custom-prop-types';

const makeMenuItems = (orgSnippets, onClick) => {
  const orgNames = Object.keys(orgSnippets);
  if (!orgNames.length) {
    return null;
  }
  return orgNames.map(orgName => (
    <SnippetList
      onClick={onClick}
      primaryText={orgName}
      snippetsList={orgSnippets[orgName]}
      snippetOwner={orgName}
    />
  ));
};

const OrgSnippetsMenu = ({ onClick, orgSnippets }) => {
  const menuItems = makeMenuItems(orgSnippets, onClick);
  return (
    <MenuItem
      disabled={!menuItems}
      menuItems={menuItems}
      primaryText="Org snippets"
      rightIcon={<ArrowDropRight />}
    />
  );
};

OrgSnippetsMenu.propTypes = {
  onClick: PropTypes.func.isRequired,
  orgSnippets: CustomPropTypes.orgSnippets.isRequired,
};

export default OrgSnippetsMenu;
