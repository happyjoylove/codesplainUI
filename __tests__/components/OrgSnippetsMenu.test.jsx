import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import OrgSnippetsMenu from '../../src/components/menus/OrgSnippetsMenu';

const defaultProps = {
  onClick: jest.fn(),
  orgSnippets: {},
};

describe('<OrgSnippetsMenu />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    it('matches when no snippets are passed', () => {
      const wrapper = shallowWithContext(
        <OrgSnippetsMenu
          {...defaultProps}
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('matches when snippets are passed', () => {
      const orgSnippets = {
        org1: {
          foo: { snippetTitle: 'bar' },
          baz: { snippetTitle: 'qux' },
        },
        org2: {},
      };
      const wrapper = shallowWithContext(
        <OrgSnippetsMenu
          {...defaultProps}
          orgSnippets={orgSnippets}
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
