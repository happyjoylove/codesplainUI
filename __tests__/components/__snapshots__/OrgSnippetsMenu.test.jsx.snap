exports[`<OrgSnippetsMenu /> snapshot tests matches when no snippets are passed 1`] = `
<MenuItem
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  checked={false}
  desktop={false}
  disabled={true}
  focusState="none"
  insetChildren={false}
  menuItems={null}
  primaryText="Org snippets"
  rightIcon={<NavigationArrowDropRight />} />
`;

exports[`<OrgSnippetsMenu /> snapshot tests matches when snippets are passed 1`] = `
<MenuItem
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  checked={false}
  desktop={false}
  disabled={false}
  focusState="none"
  insetChildren={false}
  menuItems={
    Array [
      <SnippetList
        onClick={[Function]}
        primaryText="org1"
        snippetOwner="org1"
        snippetsList={
            Object {
                "baz": Object {
                  "snippetTitle": "qux",
                },
                "foo": Object {
                  "snippetTitle": "bar",
                },
              }
        } />,
      <SnippetList
        onClick={[Function]}
        primaryText="org2"
        snippetOwner="org2"
        snippetsList={Object {}} />,
    ]
  }
  primaryText="Org snippets"
  rightIcon={<NavigationArrowDropRight />} />
`;
